### PR TITLE
server: add metrics for pd members health 

### PR DESCRIPTION
**Pull request by nolouch**
Periodic testing members health.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -642,3 +642,22 @@ func (s *Server) GetMemberLeaderPriority(id uint64) (int, error) {
 func (s *Server) SetLogLevel(level string) {
 	s.cfg.Log.Level = level
 }
+
+var healthURL = "/pd/ping"
+
+// CheckHealth checks if members are health
```
Comment (Connor1996) [inline server/server.go]: s/health/healthy

File: server/server.go [hunk 2]
```diff
@@ -642,3 +642,22 @@ func (s *Server) GetMemberLeaderPriority(id uint64) (int, error) {
...
+func (s *Server) CheckHealth(members []*pdpb.Member) map[uint64]*pdpb.Member {
+	unhealthMembers := make(map[uint64]*pdpb.Member)
+	for _, member := range members {
+		for _, cURL := range member.ClientUrls {
+			resp, err := DialClient.Get(fmt.Sprintf("%s%s", cURL, healthURL))
+			if err != nil || resp.StatusCode != http.StatusOK {
+				unhealthMembers[member.GetMemberId()] = member
+			}
+			if resp != nil {
```
Comment (Connor1996) [inline server/server.go]: as document shown, resp.Body never to be nil

File: server/server.go [hunk 3]
```diff
@@ -642,3 +642,22 @@ func (s *Server) GetMemberLeaderPriority(id uint64) (int, error) {
...
+var healthURL = "/pd/ping"
+
+// CheckHealth checks if members are health
+func (s *Server) CheckHealth(members []*pdpb.Member) map[uint64]*pdpb.Member {
+	unhealthMembers := make(map[uint64]*pdpb.Member)
+	for _, member := range members {
+		for _, cURL := range member.ClientUrls {
+			resp, err := DialClient.Get(fmt.Sprintf("%s%s", cURL, healthURL))
+			if err != nil || resp.StatusCode != http.StatusOK {
+				unhealthMembers[member.GetMemberId()] = member
```
Comment (Connor1996) [inline server/server.go]: so as long as one client urls disconnected, we think this member is unhealthy? if so, no need iterate rest, just break.

File: server/server.go (hunk 2 above)
Comment (nolouch) [reply server/server.go]: resp is nill in here when I run the test.

**Pull request merged**